### ISSUE TITLE
Using loadData on an object data doesn't work

### ISSUE THREAD

**Issue by AMBudnik** [handsontable/handsontable#4204]
### Description

When we want to use `loadData` using an object data source table shows only row headers.

![gif](https://cloud.githubusercontent.com/assets/10757813/25218629/ce9608e2-25ab-11e7-9931-012e5ff807e7.gif)

### Tip
It doesn't work only if we call the method for the first time http://jsfiddle.net/handsoncode/k6xj3uh2/

### Demo

http://jsfiddle.net/hmk4dsnk/

### Your environment
* Handsontable version: 0.31.2 / 1.10.2
* Browser Name and version: Chrome 57
* Operating System: Windows 10

**Comment by nobol** [comment]
Interestingly, if called multiple times, like this: 

```
setTimeout(function() {
    hot3.loadData(objectData);
    hot3.loadData(objectData);
    hot3.loadData(objectData);
  }, 2000) 
```

it works. We have the same issue currently.

**Comment by AMBudnik** [comment]
That is correct @nobol it only fails for the first time that we call it.
It looks like some calculations aren't finished in the time that we want to update the data.

**Comment by nobol** [comment]
I am using this as a workaround for now (calling loadData two times). I hope that also works for you!

**Comment by AMBudnik** [comment]
yes it does @nobol Still it is something to fix :)

**Comment by ericjames94** [comment]
Have there been any update to this issue?

I am using the most recent release (version 0.34.5) and I am currently having a similar problem where, when loading data into the table from an object, the correct number of rows are being created, but only the first row is populated with data.

Calling loadData twice (as mentioned above by @nobol) does not work, but my current work around is to load the data from the object, clear the data from the table (using loadData([])) and then load the data from the object again.

**Comment by AMBudnik** [comment]
Hi @ericjames94 

If you add the `columns` settings the object data is visible http://jsfiddle.net/handsoncode/bt3mxv4r/

**Comment by AMBudnik** [comment]
Still replicable in v0.35.1. Updated demo http://jsfiddle.net/czr7nyL5/

**Comment by AMBudnik** [comment]
works the same in v 5.0.1 http://jsfiddle.net/sa2ve4p8/

**Comment by felixOvianet** [comment]
The issue is still present in the 7.1.0 http://jsfiddle.net/n3ouz5kp/

**Comment by wojciechczerniak** [comment]
Looks promising in `v8.0.0-rc1`: https://jsfiddle.net/4sbft6v3/

@AMBudnik @aninde

**Comment by AMBudnik** [comment]
Yup, totally fixed for this use case.

**Comment by AMBudnik** [comment]
Thank you, solved http://jsfiddle.net/sgzp310a/

@felixOvianet does it work for you as well, and you @ericjames94 @nobol ?

**Comment by felixOvianet** [comment]
Yup 100% fine, thank you!

**Comment by Chris25602** [comment]
ran into this today as well ... looping on a timeout and trying to load does work, it just doesn't seem like that should be the solution but alas pick your battles this might not be worth it.

**Comment by AMBudnik** [comment]
I also wanted to update this topic with some examples from `7.3.0` (the latest version).

## Starting with object data 
#### Main demo https://jsfiddle.net/qs2au4yr

#### Case 1
Using the same `keys` in object. Update `object` to `object`
1. Click <kbd>Add object data</kbd>
2. You get `Adam`, `null` in columns (wrong)
3. Click the same button again
4. You get `1`, `Adam` (correct)
![obj](https://user-images.githubusercontent.com/10757813/71165040-68406480-2250-11ea-98ec-6dc8c4a755ed.gif)

#### Case 2
Using different `keys`. Update `object` to `object`.
I guess that the `name` key is held so if we change it to `name2` in the `objectData` we should get a different result. 
1. Use this demo instead https://jsfiddle.net/9ub1psgz/
2. Click the <kbd>Add object data</kbd> button and check the console result - you get `null`, `null`. (wrong)
3. Click the same button again
4. You get `1`, `Adam` (correct)

#### Case 3
Add some data. Update `object` to `object`.
1. Use this demo https://jsfiddle.net/9ub1psgz/
2. Click the <kbd>Add object data</kbd> button and check the console result - you get `null`, `null`. (wrong)
3. Type anything in the first cell
4. Click the <kbd>Add object data</kbd> button again.
5. Now there are 3 columns.

![load](https://user-images.githubusercontent.com/10757813/71165673-93778380-2251-11ea-9083-2413ded010ea.gif)

#### Case 4
Using different `keys`. Update `object` to `array`. (Steps like Case 2)
1. Use this demo https://jsfiddle.net/9ub1psgz/
2. Click the <kbd>Add array data</kbd> button and check the console result - you get `null`, `null`. (wrong)
3. Click the same button again
4. You get `100`, `Hello` (correct)

#### Case 5
It has to end up bad.
1. Go to https://jsfiddle.net/9ub1psgz/
2. Click the <kbd>Add object data</kbd> button
3. Click the <kbd>Add array data</kbd> button
4. Type anything in the first cell
5. Click the <kbd>Add object data</kbd> button
6. All data is gone. We are left with row headers.

![bug](https://user-images.githubusercontent.com/10757813/71166075-40520080-2252-11ea-96e1-f83a3d04398f.gif)

However, if you change the button in step `6` everything's OK.


## Starting with array data 
#### Main demo https://jsfiddle.net/2ua19nmp/

#### Case 1
Changing `array` to `object`.
1. Click the <kbd>Add object data</kbd> button
2. There's only a single row header and no data (check the console to see `[null]`
3. Click the same button again
4. Everything's OK now.

#### Case 2
Changling `array` to `array`
1. Click the <kbd>Add object data</kbd> button
2. Works like a charm!

### Case 3
1. Click the <kbd>Add object data</kbd> button
2. Click the <kbd>Add array data</kbd> button
3. Type anything in the first cell
4. Click the <kbd>Add object data</kbd> button
5. All data is gone. We are left with row headers.

**Comment by AMBudnik** [comment]
### Here's an update for v 8.0.0

#### The main issue solved - http://jsfiddle.net/L05az72j/
![solved](https://user-images.githubusercontent.com/10757813/89864636-8f1b2480-dbac-11ea-8927-0940dfb6aff5.gif)

#### Staring with object data
1. *  [x] Case 1 - https://jsfiddle.net/0xohzt6n/ - **solved** (we get a one click change)
1. *  [x] Case 2 - https://jsfiddle.net/vr0yqbL3/ - **solved** (correct logs from the beginning) 
1. *  [x] Case 3 - https://jsfiddle.net/fmu1gtsr/ - **solved** - correct logs and data)
1. *  [x] Case 4 - https://jsfiddle.net/8sfjowq5/ - **solved** - (correct logs)
1. *  [x] Case 5 - https://jsfiddle.net/6mgLyh1w/ - **solved** - (correct data)

#### Staring with array data - https://jsfiddle.net/4x51sm0w/
1. *  [x] Case 1 - **solved** - (data is visible from the beginning) 
1. *  [x] Case 2 - **solved** - (still works well)
1. *  [x] Case 3 - **solved** - (all, correct data is visible)

All issues are gone using v 8.0.0

**Comment by AMBudnik** [comment]
@niklaszantner @Chris25602 @felixOvianet @ericjames94

All the mentioned cases are officially solved (without workarounds) for v 8.0.0.
I highly encourage you to test the new version and leave some feedback.

CDN https://cdn.jsdelivr.net/gh/handsontable/handsontable@latest/dist/handsontable.full.css
https://cdn.jsdelivr.net/gh/handsontable/handsontable@latest/dist/handsontable.full.js
NPM npm i handsontable
Release Notes https://github.com/handsontable/handsontable/releases
